install-deps.sh

```diff
@@ -400,11 +401,28 @@ EOF
 		if test $ID = centos -a $MAJOR_VERSION = 8 ; then
                     # Enable 'powertools' or 'PowerTools' repo
                     $SUDO dnf config-manager --set-enabled $(dnf repolist --all 2>/dev/null|gawk 'tolower($0) ~ /^powertools\s/{print $1}')
+		    case "$ARCH" in
+			x86_64)
+			    $SUDO dnf -y install centos-release-scl
+			    dts_ver=10
+			    ;;
+			aarch64)
+			    $SUDO dnf -y install centos-release-scl-rh
+			    $SUDO dnf config-manager --disable centos-sclo-rh
+			    $SUDO dnf config-manager --enable centos-sclo-rh-testing
+			    dts_ver=10
+			    ;;
+		    esac
 		    # before EPEL8 and PowerTools provide all dependencies, we use sepia for the dependencies
                     $SUDO dnf config-manager --add-repo http://apt-mirror.front.sepia.ceph.com/lab-extras/8/
                     $SUDO dnf config-manager --setopt=apt-mirror.front.sepia.ceph.com_lab-extras_8_.gpgcheck=0 --save
                     $SUDO dnf -y module enable javapackages-tools
                 elif test $ID = rhel -a $MAJOR_VERSION = 8 ; then
+                    $SUDO dnf config-manager \
+			  --enable rhel-server-rhscl-8-rpms \
+			  --enable rhel-8-server-optional-rpms \
+			  --enable rhel-8-server-devtools-rpms
```

Comment:
to my knowledge, these repos do not exist:

```
pdonnell@vossi04 ~/ceph$ ./install-deps.sh 
Using dnf to install dependencies
Updating Subscription Management repositories.
Last metadata expiration check: 0:03:16 ago on Mon 15 Aug 2022 05:02:19 PM UTC.
Package yum-utils-4.0.21-11.el8.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Updating Subscription Management repositories.
Error: No matching repo to modify: rhel-8-server-devtools-rpms, rhel-server-rhscl-8-rpms, rhel-8-server-optional-rpms.
```

For `rhel-server-rhscl-8-rpms`, this should definitely not exist according to

https://access.redhat.com/support/policy/updates/rhscl

> Starting with Red Hat Enterprise Linux 8, the content traditionally consumed via Software Collections is now part of Application Streams. Please see the Application Streams Life Cycle documentation for that release.

However,  if I remove these repos from `install-deps.sh`, I get this:

> CMake Error at src/CMakeLists.txt:119 (message):
>  C++20 support requires a minimum GCC version of 11.

This is on RHEL 8.6 (`vossi04.front.sepia.ceph.com`). I recently converted this machine from sepia's satellite to the RH CDN, following [1] at David Galloway's suggestion. Am I missing something?

[1] https://access.redhat.com/solutions/4525751

Reply:
i accidentally reintroduced those rhel packages as part of a revert commit 65b1a13139873c1525456e05cd00aeb64da4e881. i didn't have a way to test against rhel, sorry. i've raised https://github.com/ceph/ceph/pull/47609 to remove them

according to ceph.spec.in, rhel should be installing the same `gcc-toolset-11` packages that centos 8 does

> However, if I remove these repos from install-deps.sh, I get this:

install-deps.sh should be giving you a message like this:
> Your GCC is too old. Please run following command to add DTS to your environment:
>
> scl enable gcc-toolset-$dts_ver bash
>
> Or add the following line to the end of ~/.bashrc and run "source ~/.bashrc" to add it permanently:
>
> source scl_source enable gcc-toolset-$dts_ver

do those steps work under rhel?

Reply:
> i accidentally reintroduced those rhel packages as part of a revert commit [65b1a13](https://github.com/ceph/ceph/commit/65b1a13139873c1525456e05cd00aeb64da4e881). i didn't have a way to test against rhel, sorry. i've raised #47609 to remove them
> 
> according to ceph.spec.in, rhel should be installing the same `gcc-toolset-11` packages that centos 8 does
> 
> > However, if I remove these repos from install-deps.sh, I get this:
> 
> install-deps.sh should be giving you a message like this:
> 
> > Your GCC is too old. Please run following command to add DTS to your environment:
> > scl enable gcc-toolset-11 bash
> > Or add the following line to the end of ~/.bashrc and run "source ~/.bashrc" to add it permanently:
> > source scl_source enable gcc-toolset-11
> 
> do those steps work under rhel?

I did not get that message. Adding `scl enable gcc-toolset-11 bash` did the trick though!